modules/api/src/main/java/org/apache/ignite/sql/BatchedArguments.java

```diff
@@ -18,24 +18,25 @@
 package org.apache.ignite.sql;
 
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
 
 /**
  * Arguments for batch query execution.
- *
- * <p>TODO: replace inheritance with delegation.
- * TODO: add arguments length validation.
- * TODO: add named arguments support.
  */
-public class BatchedArguments extends ArrayList<List<Object>> implements List<List<Object>> {
+public final class BatchedArguments implements Iterable<List<Object>> {
```

Comment:
We have size and positional access, let's implement `List` interface.

Reply:
I'm not sure about this. This class is a part of public API.
java.util.List contains ~30 methods, that are not required by this container.
I think it's better to narrow down the functionality of this class and extend it later if needed.

Reply:
Ok.